projects/plugins/jetpack/extensions/blocks/ai-assistant/extensions/ai-assistant/with-ai-assistant.tsx

```diff
@@ -72,31 +72,22 @@ export const withAIAssistant = createHigherOrderComponent(
 			[ removeBlocks, updateBlockAttributes ]
 		);
 
-		const addAssistantMessage = useCallback(
+		const updateStoredPrompt = useCallback(
 			( assistantContent: string ) => {
 				setStoredPrompt( prevPrompt => {
-					/*
-					 * Add the assistant messages to the prompt.
-					 * - Preserve the first item of the array (`system` role )
-					 * - Keep the last 4 messages.
-					 */
-
-					// Pick the first item of the array.
-					const firstItem = prevPrompt.messages.shift();
-
 					const messages: Array< PromptItemProps > = [
-						firstItem, // first item (`system` by default)
-						...prevPrompt.messages.splice( -3 ), // last 3 items
+						/*
+						 * Do not store `system` role items,
+						 * and preserve the last 3 ones.
+						 */
+						...prevPrompt.messages.filter( _ => _.role !== 'system' ).slice( -3 ),
```

Comment:
```suggestion
						...prevPrompt.messages.filter( message => message.role !== 'system' ).slice( -3 ),
```

`_` reminds me of lodash